### PR TITLE
Student's dashboard statistics

### PR DESCRIPTION
**Pull request by realARCHaim**
- Student's dashboard statistics (https://trello.com/c/oSzufYkc/75-students-dashboard-statistics)
- Default question vote fix (includes model fix and migraiton)

### CODE REVIEW

File: askup/utils/general.py [hunk 1]
```diff
@@ -55,14 +55,68 @@ def get_user_questions_count(user_id):
...
         return cursor.fetchone()[0] or 0
 
     return 0
 
 
+def get_user_place_in_rank_list(user_id):
+    """
+    Return a rank list place of the user by id.
+    """
+    with connection.cursor() as cursor:
```
Comment (cmltaWt0) [inline askup/utils/general.py:70]: Why not use Django aggregation, annotation?

https://docs.djangoproject.com/en/2.0/topics/db/aggregation/

File: askup/views.py [hunk 2]
```diff
@@ -229,19 +234,29 @@ def get_queryset(self):
...
-            'own_score': get_user_score_by_id(user.id),
-            'is_owner': user.id == request.user.id,
-            'is_student': check_user_has_groups(request.user, 'student')
+            'first_name': profile_user.first_name,
+            'last_name': profile_user.last_name,
+            'email': profile_user.email,
+            'own_profile': int(user_id) == request.user.id,
+            'own_score': get_user_score_by_id(profile_user.id),
+            'is_owner': profile_user.id == request.user.id,
+            'is_student': check_user_has_groups(request.user, 'student'),
```
Comment (cmltaWt0) [inline askup/views.py:248]: Is it still used?
And why you check group for user from request but not from profile?

File: askup/views.py [hunk 3]
```diff
@@ -229,19 +234,29 @@ def get_queryset(self):
...
+            'own_correct_answers': get_user_correct_answers_count(profile_user.id),
```
Comment (cmltaWt0) [inline askup/views.py:249]: Why not use `user_id` ?

File: askup/utils/general.py (hunk 1 above)
Comment (realARCHaim) [reply askup/utils/general.py:70]: I didn't find the "rank" function in a django aggregations documentation, - how will this data be aggregated other than by this function wrapped up into the derived table?

File: askup/views.py (hunk 3 above)
Comment (realARCHaim) [reply askup/views.py:249]: That's right, I can. But I'll need to cast it into an int type, cause `user_id` is getting here as a string.

File: askup/views.py (hunk 2 above)
Comment (realARCHaim) [reply askup/views.py:248]: You right. 

- `245 line` - is a logical duplication for the `247 line` - I'll remove it and will use an old 247 line in a template
- `248 line` - will be used in the next two parts of the Student's dashboard (the "Rank list" and the "My questions")

Thank you.

File: askup/models.py [hunk 4]
```diff
@@ -372,8 +374,12 @@ def vote(self, user_id, value):
             question_id=self.id,
             voter_id=user_id,
         )
-        vote_value = self.vote_value + value
-        return vote_value, 'Thank you for your vote!'
+        self.vote_value = F('vote_value') + value
```
Comment (cmltaWt0) [inline askup/models.py:377]: I have an suspicion that you vote twice. In Vote.save and here. But finally there is a correct result.

File: askup/utils/general.py [hunk 5]
```diff
@@ -243,3 +297,75 @@ def parse_response_url_to_parameters(response):
...
+
+def get_student_last_week_correct_answers_count(user_id):
+    """Return last week correct answers of the student."""
+    with connection.cursor() as cursor:
+        cursor.execute(
+            """
+                select count(id)
+                from askup_answer
+                where
+                    self_evaluation = 2 and
```
Comment (cmltaWt0) [inline askup/utils/general.py:345]: TODO: refactor to avoid hardcode.

**Review by cmltaWt0 — APPROVED** [review]

**Pull request merged**